libvast/include/vast/data.hpp

```diff
@@ -249,6 +251,73 @@ void merge(const record& src, record& dst,
 /// @param rhs The RHS of the predicate.
 bool evaluate(const data& lhs, relational_operator op, const data& rhs);
 
+/// Tries to find the entry with the dot-sperated `path`.
+/// @pre `!path.empty()`
+template <typename T>
+  requires caf::detail::tl_contains<data::types, T>::value
+const T* get_if(const record* r, std::string_view path) {
+  VAST_ASSERT(!path.empty());
+  auto names = detail::split(path, ".");
+  VAST_ASSERT(!names.empty());
+  auto current = r;
+  for (auto& name : names) {
+    auto last = &name == &names.back();
+    auto it = current->find(name);
+    if (it == current->end())
+      // Field not found.
+      return nullptr;
+    auto& field = it->second;
+    if (last) {
+      // Path was completely processed.
+      auto f = detail::overload{
+        [](const T& result) {
+          return &result;
+        },
+        [](const auto&) -> const T* {
+          // Found, but type does not match.
+          return nullptr;
+        },
+      };
+      return caf::visit(f, field);
```

Comment:
You could actually use `if (const auto* value = caf::get_if<T>(field)) { ... }` here to save a few lines and a few template instantiations, but I don't mind either way. The current form is more easily extensible if we ever want to support casting here.